### ISSUE TITLE
(some?) eval expressions have issues with :=

### ISSUE THREAD

**Issue by eantonya** [Rdatatable/data.table#1181]
``` R
dt1 = data.table(a = 1, key = 'a')
dt2 = data.table(c = 1, fn = list(quote(5*a)), key = 'c')

# this works
dt1[, eval(dt2[a]$fn[[1]], .SD)]
[1] 5

# this doesn't
dt1[, n := eval(dt2[a]$fn[[1]], .SD)]
#Error in eval(expr, envir, enclos) : object 'a' not found
```

Relevant SO post - http://stackoverflow.com/a/30899285/817778

**Comment by jangorecki** [comment]
This is very similar to #1180 but looks to be resolved already
```r
dt1 = data.table(a = 1, key = 'a')
dt2 = data.table(c = 1, fn = list(quote(5*a)), key = 'c')
dt1[, n := eval(dt2[a]$fn[[1]], .SD)][] 
#       a     n
#1:     1     5
```
To be closed by adding unit test to prevent future regression.

**Comment by mattdowle** [comment]
Thanks @eantonya. Looks to have been fixed in v1.9.8 (Nov 2016), bug fix 16 for #880 : 
> 16. `DT[, LHS := RHS]` with RHS is of the form `eval(parse(text = foo[1]))` referring to columns in `DT` is now handled properly. Closes [#880](https://github.com/Rdatatable/data.table/issues/880). Thanks to tyner.

PR #4361 adds the test in this issue for completeness.
